src/Network.cpp

```diff
@@ -860,9 +861,40 @@ Network::Netresult Network::get_scored_moves(
     if (ensemble == DIRECT) {
         assert(rotation >= 0 && rotation <= 7);
         result = get_scored_moves_internal(state, planes, rotation);
+    } else if (ensemble == MULTI_AVG && rotation > 1) {
+        // MULTI_AVG rotation is the # of random rotations to provide
+        assert(rotation >= 1 && rotation <= 8);
+        std::vector<int> r_list{0, 1, 2, 3, 4, 5, 6, 7};
+        // Need to account for extra 2 in m_squaresize.
+        float policy[(BOARD_SIZE + 2) * (BOARD_SIZE + 1)] = {0};
+
+        if (rotation < 8) {
+            std::random_device rd;
+            std::mt19937 g(rd());
+            std::shuffle(r_list.begin(), r_list.end(), g);
+        }
+
+        result = get_scored_moves_internal(state, planes, r_list[0]);
+
+        for (int r = 1; r < rotation; ++r) {
+            Netresult tmpresult = get_scored_moves_internal(state, planes, r_list[r]);
+            result.second += tmpresult.second;
+
+            for (auto res : tmpresult.first) {
+                // +1 to account for PASS
+                policy[res.second + 1] += res.first;
+            }
+        }
+
+        for (size_t c = 0; c < result.first.size(); c++) {
+            result.first[c].first += policy[result.first[c].second + 1];
+            result.first[c].first /= rotation;
```

Comment:
Dividing by rotation looks suspect.

Reply:
As far as I can see it looks like it is taking the average of the random rotations but it isn't clear from the name of the variable.

Reply:
I did use the variable two ways. For DIRECT rotation is the rotation #. For MULTI_AVG rotation is the # of rotations. Maybe that was silly of me.

Reply:
I think it would be clearer if we just used 8 everywhere. 😊

Reply:
Yeah same variable different meaning is pretty evil :-)

Reply:
Should I make a new variable earlier on with a different name for readability?

Reply:
I'd use num_rotations or rotations_cnt or similar when "rotation" is really "number of rotations".